### PR TITLE
Bump  to 0.7.x

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Bump  to 0.7.x

Kotlin 2, Compose Compiler from Jetbrains

#### WHY

Get ready for active development against Compose 1.8.0-alpha and Wear Compose 1.5.-alpha 

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

**Review by luizgrp — APPROVED** [review]
Thanks for taking this work 👍

File: media/sample/src/main/java/com/google/android/horologist/mediasample/ui/util/JankPrinter.kt [hunk 1]
```diff
@@ -44,7 +44,8 @@ class JankPrinter {
 
             stateHolder = PerformanceMetricsState.getHolderForHierarchy(contentView).apply {
                 state?.putState("Activity", activity.javaClass.simpleName)
-                state?.putState("route", NavigationScreens.Player.navRoute)
+                // TODO compute this, as it is probably wrong
```
[the remark this answers is not in the document]
Comment (yschimke) [reply media/sample/src/main/java/com/google/android/horologist/mediasample/ui/util/JankPrinter.kt:47]: I'll leave blank for now, removing this line.

File: media/sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/UampSettingsScreen.kt [hunk 2]
```diff
@@ -78,14 +79,20 @@ fun UampSettingsScreen(
...
                         enabled = !screenState.guestMode,
                     )
                 } else {
                     Chip(
                         label = stringResource(id = R.string.logout),
                         modifier = Modifier.fillMaxWidth(),
-                        onClick = { navController.navigateToGoogleSignOutScreen() },
+                        onClick = {
+                            navController.navigate(GoogleSignOutScreen) {
+                                popUpTo<NavigationScreen.Player>()
```
[the remark this answers is not in the document]
Comment (yschimke) [reply media/sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/UampSettingsScreen.kt:93]: Not sure, I haven't seen that pattern generally.

https://github.com/android/nowinandroid/blob/420076f01aadd61f650b6848cd23643ca59226b8/app/src/main/kotlin/com/google/samples/apps/nowinandroid/ui/interests2pane/InterestsListDetailScreen.kt#L125

[the remark this answers is not in the document]
Comment (yschimke) [reply media/sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/UampSettingsScreen.kt:93]: Actually I have two in MediaNavController

[the remark this answers is not in the document]
Comment (yschimke) [reply media/sample/src/main/java/com/google/android/horologist/mediasample/ui/settings/UampSettingsScreen.kt:93]: But that's not sample code, so leaving.

**Pull request merged**